modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1235,29 +1234,48 @@ def _compute_map_reduce_metadata(self, axis, new_parts):
         )
         return result
 
-    def fold_reduce(self, axis, func):
+    def reduce(
+        self,
+        axis: int,
+        function: Callable,
+        dtypes: Optional[str] = None,
+    ) -> "PandasDataframe":
         """
-        Apply function that reduces Frame Manager to series but requires knowledge of full axis.
+        Perform a user-defined per-column aggregation, where each column reduces down to a single value.
 
         Parameters
         ----------
-        axis : {0, 1}
-            The axis to apply the function to (0 - index, 1 - columns).
-        func : callable
-            The function to reduce the Manager by. This function takes in a Manager.
+        axis : int
+            The axis to perform the reduce over.
+        function : callable
+            The reduce function to apply to each column.
+        dtypes : str, default: None
+            The data types for the result. This is an optimization
+            because there are functions that always result in a particular data
+            type, and this allows us to avoid (re)computing it.
 
         Returns
         -------
         PandasDataframe
             Modin series (1xN frame) containing the reduced data.
+
+        Notes
+        -----
+        The user-defined function must reduce to a single value.
         """
-        func = self._build_mapreduce_func(axis, func)
+        function = self._build_treereduce_func(axis, function)
         new_parts = self._partition_mgr_cls.map_axis_partitions(
-            axis, self._partitions, func
+            axis, self._partitions, function
         )
-        return self._compute_map_reduce_metadata(axis, new_parts)
+        return self._compute_tree_reduce_metadata(axis, new_parts)
 
-    def map_reduce(self, axis, map_func, reduce_func=None):
+    def tree_reduce(
+        self,
+        axis: int,
+        map_func: Callable,
+        reduce_func: Optional[Callable] = None,
+        dtypes: Optional[str] = None,
+    ) -> "PandasDataframe":
```

Comment:
Signature of `ModinDataframe.tree_reduce` is different from this. Is this expected?